sdk-cpp/src/do_download.cpp

```diff
@@ -58,23 +58,61 @@ void download::download_url_to_path(const std::string& uri, const std::string& d
     download_url_to_path(uri, downloadFilePath, isCancelled, timeOut);
 }
 
+// Ensures download is finalized/aborted in the agent to avoid stale download objects.
+// Future: We could expect callers to explicitly opt-in to persist downloads in the agent and
+//   use that as an indicator to perform/skip the cleanup directly in ~download().
+class download_cleanup_call
```

Comment:
Does adding this cleanup impact the dosvc client? If so, consider adding this logic to the specific download_impl class

Reply:
This is required behavior for both windows and xplat.